### DISCUSSION TITLE
Retrieving results

### DISCUSSION THREAD

**Discussion by aphilas** [riverqueue/river#500] [Q&A] [answered]
I might be missing something obvious here, how can I retrieve the results of a job?

 The equivalent in Oban should be [`fetch_recorded/1`](https://getoban.pro/docs/pro/1.2.1/Oban.Pro.Worker.html#module-retrieving-results).

**Comment by bgentry** [comment]
There’s no feature for recording job “output” as of now. I’ve definitely been thinking about how to add it though. Any thoughts on how the API for that might look? Also how large is the payload you’re looking to save?

**Reply by aphilas** [reply]
Unfortunately I haven't thought about how it should work/look like.


> Also how large is the payload you’re looking to save?

Write now it's anything from a few hundred bytes to a few kilobytes of json, but it may grow larger depending on the input.

**Comment by travisby** [comment, marked as answer]
On the subject,

`RecordOutput` has existed since v0.18.0 - https://pkg.go.dev/github.com/riverqueue/river#RecordOutput

and `JobUpdate` since v0.29.0 - https://pkg.go.dev/github.com/riverqueue/river#Client.JobUpdate

I think this covers what I've been looking for -- a way to look up at some past information on a completed job!
